src/content/datachannel/basic/js/main.js

```diff
@@ -64,7 +64,11 @@ function createConnection() {
   remoteConnection.onicecandidate = iceCallback2;
   remoteConnection.ondatachannel = receiveChannelCallback;
 
-  localConnection.createOffer(gotDescription1, onCreateSessionDescriptionError);
+  localConnection.createOffer().then(
+    gotDescription1
+  ).catch(
+    onCreateSessionDescriptionError
```

Comment:
You wanna use `.then(onSuccess, onFailure)` rather than `.then(onSuccess).catch(onFailure)` because the latter incorrectly catches any coding errors in `onSuccess` which you don't want.